staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go

```diff
@@ -159,15 +185,53 @@ func newTestableController(
 		serverConcurrencyLimit: serverConcurrencyLimit,
 		requestWaitLimit:       requestWaitLimit,
 		flowcontrolClient:      flowcontrolClient,
+		id:                     makeID(),
 		priorityLevelStates:    make(map[string]*priorityLevelState),
 	}
-	klog.V(2).Infof("NewTestableController with serverConcurrencyLimit=%d, requestWaitLimit=%s", serverConcurrencyLimit, requestWaitLimit)
+	klog.V(2).Infof("NewTestableController with id=%q, serverConcurrencyLimit=%d, requestWaitLimit=%s", cfgCtl.id, serverConcurrencyLimit, requestWaitLimit)
 	cfgCtl.initializeConfigController(informerFactory)
 	// ensure the data structure reflects the mandatory config
 	cfgCtl.lockAndDigestConfigObjects(nil, nil)
 	return cfgCtl
 }
 
+// makeID computes the ID to use for this apiserver in
+// PriorityLevelConfigurationStatus::ConcurrencyLimits.  If possible
+// it will be based on `net.InterfaceAddrs`, consisting of a hash of
+// all of them plus one of the distinctive ones.  If that is not
+// possible then the ID will be a timestamp of maximum precision.
+func makeID() string {
```

Comment:
can we reuse `--advertise-address`?

https://github.com/kubernetes/kubernetes/blob/d682c8389087e55835bbd02ce20cb6110c81731f/staging/src/k8s.io/apiserver/pkg/server/options/server_run_options.go#L158-L162

Reply:
Oh yes, that was suggested.  Just have to see how far it is plumbed.

Reply:
I see that option passed into https://github.com/kubernetes/apiserver/blob/32590dd629be09b92d2a9f224d42b6ed61858de8/pkg/server/config.go#L219 --- which is in the section of "targets for removal".  I also see a conditional propagation to https://github.com/kubernetes/apiserver/blob/32590dd629be09b92d2a9f224d42b6ed61858de8/pkg/server/config.go#L134 --- which looks to me like it sometimes might be set to a load balancer's ingress address.  So I think it might be best to leave the code as it is.

Reply:
Actually, I think the suggestion was more like using the existing code to pick an address.  That is invoked by https://github.com/kubernetes/apiserver/blob/32590dd629be09b92d2a9f224d42b6ed61858de8/pkg/server/options/serving.go#L120 when there is no bind address.

Reply:
Latest revision takes that suggestion.

Reply:
I haven't looked yet, but you shouldn't use bind address for the identity if that's why it's being talked about-- you should use the identity @roycaihw added with https://github.com/kubernetes/kubernetes/pull/95428 and https://github.com/kubernetes/kubernetes/pull/95533

Reply:
Oh, I see this is super old -- sorry if my comment is not relevant anymore!